packages/analytics-js-plugins/src/errorReporting/utils.ts

```diff
@@ -0,0 +1,192 @@
+};
+
+const getURLWithoutQueryString = () => {
+  const url = globalThis.location.href.split('?');
+  return url[0];
+};
+
+const getErrorContext = (event: any) => {
+  const { message } = event;
+  let context = message;
+
+  // Hack for easily grouping the script load errors
+  // on the dashboard
+  if (message.includes('Error in loading a third-party script')) {
+    context = 'Script load failures';
+  }
+  return context;
+};
+
+const getBugsnagErrorEvent = (
+  payload: ErrorFormat,
+  errorState: ErrorState,
+  state: ApplicationState,
+): ErrorEventPayload => ({
+  notifier: {
+    name: NOTIFIER_NAME,
+    version: '__PACKAGE_VERSION__',
+    url: SDK_GITHUB_URL,
+  },
+  events: [
+    {
+      payloadVersion: '5',
+      exceptions: clone(payload.errors),
+      severity: errorState.severity,
+      unhandled: errorState.unhandled,
+      severityReason: errorState.severityReason,
+      app: {
+        version: state.context.app.value.version,
+        releaseStage: getReleaseStage(),
+      },
+      device: {
+        locale: state.context.locale.value ?? undefined,
+        userAgent: state.context.userAgent.value || undefined,
+        time: new Date(),
+      },
+      request: {
+        url: getURLWithoutQueryString() as string,
+        clientIp: '[NOT COLLECTED]',
+      },
+      breadcrumbs: clone(state.reporting.breadcrumbs.value),
+      context: getErrorContext(payload.errors[0]),
+      metaData: {
+        sdk: {
+          name: 'JS',
+          installType: '__MODULE_TYPE__',
+        },
+        state: getAppStateForMetadata(state) ?? {},
+        source: {
+          id: state.source?.value?.id,
+          snippetVersion: (globalThis as typeof window).RudderSnippetVersion,
+        },
+      },
+      user: {
+        id: state.lifecycle.writeKey.value as string,
+      },
+    },
+  ],
+});
+
+const isRudderSDKError = (event: any) => {
+  const errorOrigin = event.stacktrace?.[0]?.file;
+
+  if (!errorOrigin || typeof errorOrigin !== 'string') {
+    return false;
+  }
+
+  const srcFileName = errorOrigin.substring(errorOrigin.lastIndexOf('/') + 1);
+  const paths = errorOrigin.split('/');
+  // extract the parent folder name from the error origin file path
+  // Ex: parentFolderName will be 'sample' for url: https://example.com/sample/file.min.js
+  const parentFolderName = paths[paths.length - 2];
+
+  return (
+    parentFolderName === CDN_INT_DIR ||
+    SDK_FILE_NAME_PREFIXES().some(
+      prefix => srcFileName.startsWith(prefix) && srcFileName.endsWith('.js'),
+    )
+  );
+};
+
+const getErrorDeliveryPayload = (payload: ErrorEventPayload, state: ApplicationState): string => {
+  const data = {
+    version: METRICS_PAYLOAD_VERSION,
+    message_id: generateUUID(),
+    source: {
+      name: SOURCE_NAME,
+      sdk_version: state.context.app.value.version,
+      write_key: state.lifecycle.writeKey.value as string,
+      install_type: '__MODULE_TYPE__',
```

Comment:
It should be changed similar to the other comment above.

Reply:
Addressed